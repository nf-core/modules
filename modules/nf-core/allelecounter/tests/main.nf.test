nextflow_process {

    name "Test Process ALLELECOUNTER"
    script "../main.nf"
    process "ALLELECOUNTER"

    tag "modules"
    tag "modules_nfcore"
    tag "allelecounter"

    test("homo_sapiens illumina paired-end sorted [cram]") {

        when {
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ]
                input[1] = [ file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true) ] // positions
                input[2] = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ] //fasta
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens illumina paired-end sorted [cram] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ]
                input[1] = [ file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true) ] // positions
                input[2] = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ] //fasta
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.allelecount.collect { file(it[1]).getName() } +
                                process.out.versions ).match() }
            )
        }

    }

}
