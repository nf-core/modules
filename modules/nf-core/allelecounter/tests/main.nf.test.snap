{
    "homo_sapiens illumina paired-end sorted [cram] - stub": {
        "content": [
            [
                "test.alleleCount",
                "versions.yml:md5,8bdaeec50c801607b958c025903e59eb"
            ]
        ],
        "timestamp": "2023-12-25T23:25:08.851096878"
    },
    "homo_sapiens illumina paired-end sorted [cram]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.alleleCount:md5,2f83352a185168c7c98e9e42550b2856"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8bdaeec50c801607b958c025903e59eb"
                ],
                "allelecount": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.alleleCount:md5,2f83352a185168c7c98e9e42550b2856"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8bdaeec50c801607b958c025903e59eb"
                ]
            }
        ],
        "timestamp": "2023-12-25T23:24:58.688570785"
    }
}